crates/sncast/src/main.rs

```diff
@@ -205,7 +206,8 @@ async fn run_async_command(
                 &artifacts,
                 wait_config,
             )
-            .await;
+            .await
+            .map_err(handle_starknet_command_error);
```

Comment:
you mean modify `print_command_result` so it will accept `Result<T, StarknetCommandError>`?

Reply:
this is one idea, yes - can't think of any major drawbacks atm, but if you see some - please let me know

Reply:
> Would it be possible to just return an error as a struct we need in the subcommands functions?
Could you elaborate on this one?

Reply:
it seems that you are doing:
```
.map_err(handle_starknet_command_error)
```

for the subcommands in all places they are used. Maybe it would make sense, to just change the subcommands functions to return whatever is needed, to avoid mapping the errors?

Reply:
The plan is to map StarknetCommandError onto ScriptCommandError in the next PR. So in scripts `.map_err(handle_starknet_command_error)` will disappear.

Reply:
> The plan is to map StarknetCommandError onto ScriptCommandError in the next PR. So in scripts .map_err(handle_starknet_command_error) will disappear.

This is why these subcommands return `StarknetCommandError`.

Reply:
> change the subcommands functions to return whatever is needed

So either way it is intended that they return `StarknetCommandError`

Reply:
One way would be to make all subcommands return `StarknetCommandError`,  the other one is to replace `handle_starknet_command_error` with `impl From<StarknetCommandError> for anyhow::Error` and make `print_command_result` accept Result<T, E> where E can be converted to anyhow::Error.


But I think we have a separate issue for creating Error type for sncast subcommands https://github.com/foundry-rs/starknet-foundry/issues/1606 , so maybe we deal with it in a separate PR?

Reply:
yes, we should probably think about it a bit harder and get back to this later on. I also understand that current behaviour will be altered for scripts in consecutive PRs.

I'm fine with doing so under said issue, but please link this discussion there, and provide a short summary (as lately github links to discussions tend to fail, and we later on do not remember what it was about 😅 ). The issue description could be improved as well, so it is clearer what should be done, as a side note.

Reply:
added comments to the issue